src/modules/components/proposal/proposalActions/proposalActionsTypes/proposalAction.ts

```diff
@@ -0,0 +1,60 @@
+export enum ProposalActionType {
+    WITHDRAW_TOKEN = 'withdrawToken',
+}
+
+export interface IProposalActionInputDataParameter {
+    /**
+     * The type of the parameter
+     */
+    type: string;
+    /**
+     * The value of the parameter
+     */
+    value: string;
+}
+
+export interface IProposalActionInputData {
+    /**
+     * The function to call from proposal action
+     */
+    function: string;
+    /**
+     * The contract address to call
+     */
+    contract: string;
+    /**
+     * The parameters to pass to the function
+     */
+    parameters: IProposalActionInputDataParameter[];
+}
+
+export interface IProposalAction {
+    /**
+     * The address of the contract the proposal is attempting to interact with
+     */
+    contractAddress: string;
```

Comment:
Isn't the `to` parameter already the `contractAddress`?

Reply:
Not sure I follow exactly, but thinking a bit more about it if so, I guess it would be the 'from' then maybe? As in we are proposing that a treasury contract sends money 'from' itself 'to' a recipient contract/address??

Reply:
Well, the `withdraw` action is a special case. In general, for all the actions, we're displaying the name of the smart contract the DAO is interacting with:
Add members => from: DAO to: Multisig
Mint token => from: DAO to: GovernanceToken
Not sure if we should have a special case for the `withdraw`, let's double check this with selim & evan